app.js

```diff
@@ -1,100 +1,74 @@
-  function display() {
-    bookDisplay.innerHTML = ''; // Clearing the previous instances before displaying the updated ones
-    bookHolder.forEach((book, index) => {
-      const bookInstance = document.createElement('article');
-      const dispTitle = document.createElement('h2');
-      dispTitle.textContent = book.title;
-      const dispAuthor = document.createElement('p');
-      dispAuthor.textContent = book.author;
-      const delButton = document.createElement('button');
-      delButton.textContent = 'remove';
-      delButton.setAttribute('data-index', index);
-      delButton.addEventListener('click', deleteBook);
-      const hrline = document.createElement('hr');
-      bookInstance.append(dispTitle, dispAuthor, delButton, hrline);
-      bookDisplay.append(bookInstance);
-    });
+  saveBookHolder(bookHolder) {
+    localStorage.setItem(this.bookHolderKey, JSON.stringify(bookHolder));
   }
 
-  // Removing a book from the bookHolder array and updating localStorage and the UI
-  function deleteBook(event) {
-    const index = event.target.getAttribute('data-index');
-    bookHolder.splice(index, 1);
-    localStorage.setItem('storedBooks', JSON.stringify(bookHolder));
-    display();
+  getInputData() {
+    return JSON.parse(localStorage.getItem(this.inputDataKey)) || {};
   }
 
-  // Creating an error message and displaying it when the user submits an empty form
-  const error = document.createElement('span');
-  error.textContent = 'Please fill out all fields.';
-
-  addButton.addEventListener('click', () => {
-    if (inputTitle.value === '' || inputAuthor.value === '') {
-      bookDisplay.append(error);
-    } else {
-      addBook();
-    }
-  });
-});
-
-// Creating a new book instance and appending it to the bookDisplay div
-function display() {
-  bookDisplay.innerHTML = ''; // Clearing the previous instances before displaying the updated ones
+  saveInputData(inputData) {
+    localStorage.setItem(this.inputDataKey, JSON.stringify(inputData));
+  }
+}
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
+const domElements = new DOMElements();
+const localStorageHandler = new LocalStorageHandler();
+const bookHolder = localStorageHandler.getBookHolder();
+let inputData = localStorageHandler.getInputData();
+function addBookToHolder(newBook) {
+  bookHolder.push(newBook);
+  localStorageHandler.saveBookHolder(bookHolder);
+}
+function clearInputData() {
+  inputData = { inputTitle: '', inputAuthor: '' };
+  localStorageHandler.saveInputData(inputData);
+}
+function addBook() {
+  const titleData = domElements.inputTitle.value;
+  const authorData = domElements.inputAuthor.value;
+  if (titleData === '' || authorData === '') {
+    domElements.displayError();
+    return;
+  }
+  const newBook = new Book(titleData, authorData);
+  addBookToHolder(newBook);
+  clearInputData();
+  displayBooks();
+}
```

Comment:
- [ OPTIONAL] After submission of the book, you should empty your inputs, so that user should not have to delete the the previous submitted book, before adding the new one.